src/node.rs

```diff
@@ -891,6 +905,38 @@ impl PbftNode {
         self._broadcast_message(&PbftMessageType::ViewChange, msg_bytes, state)
     }
 
+    /// Check the on-chain list of peers; if it has changed, update peers list and return true.
+    fn update_membership(&mut self, block_id: BlockId, state: &mut PbftState) -> bool {
+        // Get list of peers from settings
+        let settings = self
+            .service
+            .get_settings(
+                block_id,
+                vec![String::from("sawtooth.consensus.pbft.peers")],
+            ).expect("Failed to get settings");
+        let peers = get_peers_from_settings(&settings);
+        let new_peers_set: HashSet<PeerId> = HashSet::from_iter(peers.clone());
+
+        // Check if membership has changed
+        let old_peers_set: HashSet<PeerId> = HashSet::from_iter(state.peer_ids.iter().cloned());
+        let difference: HashSet<PeerId> = new_peers_set
```

Comment:
It looks like we don't use `difference` other than checking that it's non-empty, so we could probably just do `if new_peers_set != old_peers_set {`.